lib/segment/src/index/plain_payload_index.rs

```diff
@@ -249,10 +249,14 @@ impl PlainIndex {
         filter: Option<&Filter>,
     ) -> bool {
         let vector_storage = self.vector_storage.borrow();
+        let Some(size_of_available_vectors) = vector_storage.size_of_available_vectors_in_bytes()
+        else {
+            // In case this gets called with sparse vectors, which are always indexed.
```

Comment:
It seems like an abstraction leak. There are no guarantees in the interface of the vector storage, that if vector size is none => segment is always indexed. If we can have another type of vectors in future, which doesn't satisfy this assumption. If we assume that this situation should never happen in a Plain segment, it should be debug assertion

Reply:
Added a debug assertion